src/store/dataset.ts

```diff
@@ -320,16 +320,27 @@ class DatasetStore {
         (item: any) => item.name === name,
       )
 
-      if (condition[0] === FilterKindEnum.Enum) {
+      if (
+        condition[0] === FilterKindEnum.Enum ||
+        condition[0] === FilterKindEnum.Func
+      ) {
         condition[3]?.forEach((value: string) => {
           filterStore.addSelectedFilters({
             group: filterItem.vgroup,
             groupItemName: name,
             variant: [value, 0],
           })
         })
+      } else {
+        filterStore.addSelectedFilters({
+          group: filterItem.vgroup,
+          groupItemName: name,
+          variant: [name, condition[condition.length - 1]],
+        })
+        filterStore.setFilterCondition(name, condition[condition.length - 1])
```

Comment:
It's something wrong with setting the condition.
The condition is supposed to be an object that holds filter name and values.
Example:
`{ "Max_GQ": { min: 99, max: undefined } , "FS": { min: undefined, max:  30 }}`
Now I get this when I console.log filterCondition
<img width="496" alt="Screenshot 2022-01-26 at 02 14 43" src="https://user-images.githubusercontent.com/96476703/151075641-a750bc1d-9f4f-4dbd-8621-f71a9136be24.png">